montepy/mcnp_problem.py

```diff
@@ -390,25 +391,56 @@ def add_cell_children_to_problem(self):
         self._transforms = Transforms(transforms)
         self._data_inputs = sorted(set(self._data_inputs + materials + transforms))
 
-    def write_to_file(self, new_problem, overwrite=False):
+    def write_problem(self, destination, overwrite=False):
+        """
+        Write the problem to a file or writeable object.
+
+        :param destination: File path or writable object
+        :type destination: object
+        :param overwrite: Whether to overwrite 'destination' if it is an existing file
+        :type overwrite: bool
+        """
+        if hasattr(destination, "write") and callable(getattr(destination, "write")):
+            # TODO: May need to ensure it also has `.tell()`
+            return self.write_to_stream(destination)
+        if isinstance(destination, (str, os.PathLike)):
+            return self.write_to_file(destination, overwrite)
+        raise TypeError(
+            f"destination f{destination} is not a file path or writable object"
+        )
+
+    def write_to_file(self, file_path, overwrite=False):
         """
         Writes the problem to a file.
 
         .. versionchanged:: 0.3.0
             The overwrite parameter was added.
 
-        :param new_problem: the file name to write this problem to
-        :type new_problem: str
+        :param file_path: the file path to write this problem to
+        :type file_path: str
         :param overwrite: Whether to overwrite the file at 'new_problem' if it exists
         :type overwrite: bool
         :raises IllegalState: if an object in the problem has not been fully initialized.
         :raises FileExistsError: if a file already exists with the same path.
         :raises IsADirectoryError: if the path given is actually a directory.
         """
-        new_file = MCNP_InputFile(new_problem, overwrite=overwrite)
-        with new_file.open("w") as fh, warnings.catch_warnings(
-            record=True
-        ) as warning_catch:
+        new_file = MCNP_InputFile(file_path, overwrite=overwrite)
+        with new_file.open("w") as fh:
+            self.write_to_stream(fh)
+        # Consider returning result of write_to_stream()
+
+    def write_to_stream(self, fh):
+        """
+        Writes the problem to a writeable stream.
+
+        TODO: Expand MCNP_InputFile or adjust this method
+              to fully suppport generic file handles.
```

Comment:
I think that's the ultimate solution here. How it is getting the input file lineno is just `\n` counting, which we can totally do with a stream. 

As for `tell` I don't think it's necessary. `tell` has no correlation to line number (or even character count sometimes, and is only useful for `seek`. I don't think we should implement `seek`.